homework02/allan/string.h

```diff
@@ -0,0 +1,41 @@
+
+#pragma once
+
+#include <iosfwd>
+#include <memory>
+
+class String
+{
+public:
+    String();
+    String(const char *);
```

Comment:
`explicit`?

Reply:
I plan to do that next. I was already stung with this with my insertion operator stack overflow